d3b-center/image-deid-etl
# Build and publish container image to Amazon ECR

## Overview

- Utilize `cibuild` and `cipublish` to build a container image for the Python application and publish it to Amazon ECR.
- Provide a method for passing the image tag to the Terraform project.
- Move the `diagnosis_mapping.json` file inside the `image_deid_etl` Python module. Use `importlib.resources` to access the file such that we no longer depend on the current working directory.

Resolves #4 

### Checklist


## Testing Instructions

Make sure your AWS SSO session is current:

```console
$ export AWS_PROFILE=chopd3bprod
$ aws sso login
```

Use `cibuild` to build a tagged container image for the Python application:

```console
$ ./scripts/cibuild
...
 => exporting to image                                                                                                                                                                                 2.5s 
 => => exporting layers                                                                                                                                                                                2.5s 
 => => writing image sha256:5bbe35c544280ec4ff2ab1f7952f825e1eea37ee9c6bdea99ac11937f7632026                                                                                                           0.0s
 => => naming to docker.io/library/image-deid-etl:a1b7d6a 
```

Use `cipublish` to publish the container image to Amazon ECR:

```console
$ ./scripts/cipublish
```

Launch an instance of the included Terraform container image:

```console
$ docker-compose -f docker-compose.ci.yml run --rm terraform
bash-5.1#
```

Once inside the context of the container image, set `GIT_COMMIT` to the tag of a published container image (e.g., `a1b7d6a`):

```console
bash-5.1# export GIT_COMMIT=a1b7d6a
```

Use `infra` to generate and apply a Terraform plan:

```console
bash-5.1# ./scripts/infra plan
bash-5.1# ./scripts/infra apply
```

**Note:** At this stage, the `image_tag` variable is a noop. However, this flow will make more sense in the next pull request.
